lib/puppet/interface.rb

```diff
@@ -154,10 +154,7 @@ def initialize(name, version, &block)
     end
 
     @name    = Puppet::Interface::FaceCollection.underscorize(name)
-
-    # SemVer is deprecated but in 4.x we must use it here (the attr_reader is public api). The
-    # extra boolean argument suppresses the deprecation warning.
-    @version = SemVer.new(version, true)
+    @version = SemanticPuppet::Version.parse(version)
```

Comment:
I'm not sure we want this backported to 4.10.x as it's not i18n related? There does seem to be a perf benefit, see https://tickets.puppetlabs.com/browse/PUP-5027. /cc @thallgren